packages/repository/test/unit/decorator/relation.decorator.unit.ts

```diff
@@ -147,19 +133,21 @@ describe('getModelRelations', () => {
 
     @model()
     class Customer extends User {
-      @hasMany(Order)
+      @hasMany(() => Order)
       orders: Order[];
     }
 
     const relations = getModelRelations(Customer);
-    expect(relations).to.deepEqual({
+    expect(relations).to.containDeep({
       accessTokens: {
-        keyTo: 'userId',
```

Comment:
Is the reason we don't have `keyTo` here because the class resolution hasn't happened yet or we don't have use for it at this point?

Reply:
With the current codebase, we could infer `keyTo` inside the `@hasMany` decorator and set it in relation metadata. However, one of the next steps is to change the way how the foreign key is inferred and leverage `@belongsTo` decorator (see https://github.com/strongloop/loopback-next/pull/1618/files#diff-444ac7611514971eb1c8e5790e78035cR135). Inspecting `@belongsTo` properties on the target model is the part that cannot be done from `@hasMany` but must wait until all classes are created and the target type can be resolved.

Setting aside `@belongsTo`, I think it's still a good idea keep `keyTo` optional, because it simplifies relation configuration for people not using decorators but writing the model definition manually via `ModelDefinition` class.